src/Mainloop.cu

```diff
@@ -56,6 +56,14 @@ template <class T> void MainLoop(Param &XParam, Forcing<float> XForcing, Model<T
 		resetmeanmax(XParam, XLoop, XModel, XModel_g);
 
 		printstatus(XLoop.totaltime, XLoop.dt);
+
+		// Stop calculation if crashed
+		if (XLoop.dt < XParam.dtmin)
+		{
+			XLoop.totaltime = XParam.endtime + 1;
+			log(" \n ");
```

Comment:
Add : 
 XLoop.nextoutputtime = XLoop.totaltime

this will enforce the request for output without changing the if statement below. same effect but cleaner I think.

Reply:
I will add this too.